pkg/limiters/global_limiter.go

```diff
@@ -0,0 +1,101 @@
+package limiters
+
+import (
+	"github.com/lwolf/konsumerator/pkg/helpers/tests"
+
+	"github.com/go-logr/logr"
+	konsumeratorv1alpha1 "github.com/lwolf/konsumerator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type GlobalLimiter struct {
+	availCPU *resource.Quantity
+	availMem *resource.Quantity
+	log      logr.Logger
+}
+
+func NewGlobalLimiter(policy *konsumeratorv1alpha1.ResourcePolicy, used *corev1.ResourceList, log logr.Logger) *GlobalLimiter {
+	l := &GlobalLimiter{log: log}
+	if policy == nil || policy.GlobalPolicy == nil {
+		// no limits were set
+		return l
+	}
+	if used == nil {
+		used = tests.NewResourceList("0", "0")
+	}
+
+	limit := policy.GlobalPolicy.MaxAllowed
+	if used.Cpu().Cmp(*limit.Cpu()) == 1 || used.Memory().Cmp(*limit.Memory()) == 1 {
+		l.log.V(1).Info(
+			"cpu limit is less than used",
```

Comment:
message a bit misleading, since you're checking both cpu and memory. Maybe something like `Resource allocation is higher than global limit`